front/src/modules/apollo/optimistic-effect/hooks/useOptimisticEffect.ts

```diff
@@ -0,0 +1,48 @@
+import { useApolloClient } from '@apollo/client';
+import { useRecoilCallback } from 'recoil';
+
+import { optimisticEffectState } from '../states/optimisticEffectState';
+import { OptimisticEffect } from '../types/OptimisticEffect';
+
+export function useOptimisticEffect() {
+  const apolloClient = useApolloClient();
+
+  const registerOptimisticEffect = useRecoilCallback(
+    ({ snapshot, set }) =>
+      (optimisticEffect: OptimisticEffect<unknown, unknown>) => {
+        const { key } = optimisticEffect;
+        const optimisticEffects = snapshot
+          .getLoadable(optimisticEffectState)
+          .getValue();
+
+        set(optimisticEffectState, {
+          ...optimisticEffects,
+          [key]: optimisticEffect,
+        });
+      },
+  );
+
+  const triggerOptimisticEffects = useRecoilCallback(
+    ({ snapshot }) =>
+      (typename: string, entities: any[]) => {
```

Comment:
Would be nice to derivate a string literal from somewhere, but it's ok as a string

Reply:
I think we won't be able as we move to the flexible schema anyway!